src/JMS/Serializer/Metadata/Driver/XmlDriver.php

```diff
@@ -351,6 +367,8 @@ protected function loadMetadataFromFile(\ReflectionClass $class, $path)
             }
         }
 
+        $this->classMetadataUpdater->update($metadata);
```

Comment:
All calls are made at the end of `loadMetadataFromFile`. Shouldn't this be called somewhere in `AbstractFileDriver`? This would simplify the logic flow